src/sentry/static/sentry/app/views/settings/incidentRules/triggers/actionsPanel/index.tsx

```diff
@@ -54,47 +53,61 @@ type Props = {
   onChange: (triggerIndex: number, triggers: Trigger[], actions: Action[]) => void;
 };
 
-const getPlaceholderForType = (type: ActionType) => {
-  switch (type) {
-    case ActionType.SLACK:
-      return '@username or #channel';
-    case ActionType.MSTEAMS:
-      //no prefixes for msteams
-      return 'username or channel';
-    case ActionType.PAGERDUTY:
-      return 'service';
-    default:
-      throw Error('Not implemented');
-  }
+/**
+ * When a new action is added, all of it's settings should be set to their default values.
+ * @param actionConfig
+ */
+const getCleanAction = (actionConfig): Action => {
+  return {
+    type: actionConfig.type,
+    targetType:
+      actionConfig &&
+      actionConfig.allowedTargetTypes &&
+      actionConfig.allowedTargetTypes.length > 0
+        ? actionConfig.allowedTargetTypes[0]
+        : null,
+    targetIdentifier: '',
+    integrationId: actionConfig.integrationId,
+    options: actionConfig.options || null,
+  };
 };
+
 /**
- * Lists saved actions as well as control to add a new action
+ * Actions have a type (e.g. email, slack, etc), but only some have
+ * an integrationId (e.g. email is null). This helper creates a unique
+ * id based on the type and integrationId so that we know what action
+ * a user's saved action corresponds to.
  */
-class ActionsPanel extends React.PureComponent<Props> {
-  /**
-   * Actions have a type (e.g. email, slack, etc), but only some have
-   * an integrationId (e.g. email is null). This helper creates a unique
-   * id based on the type and integrationId so that we know what action
-   * a user's saved action corresponds to.
-   */
-  getActionUniqueKey({type, integrationId}: Pick<Action, 'type' | 'integrationId'>) {
-    return `${type}-${integrationId}`;
-  }
+const getActionUniqueKey = ({
+  type,
+  integrationId,
+}: Pick<Action, 'type' | 'integrationId'>) => {
+  return [type, integrationId].filter(x => !!x).join('-');
```

Comment:
gonna be honest here in that your change made it so that I didn't know what this was doing anymore, I don't javascript very much so maybe this is super common but I don't think the way we had it before was bad. 

I'm not saying to change it back, but just putting it out there

Reply:
You're right, I'll replace it.

This pattern is useful in a few cases when the if-else code would be really complex. It filters out nullish values and then takes care of spacing. For example:
```
const fullName = [givenName, middleName, familyName].filter(x => !!x).join(' ');
```